developer-extension/src/panel/components/tabs/eventsTab/eventsList.tsx

```diff
@@ -1,19 +1,213 @@
-import { Table } from '@mantine/core'
-import React from 'react'
-import type { EventFilters } from '../../../hooks/useEvents'
+import { Popover, Box, Text, Button, Flex, Autocomplete } from '@mantine/core'
+import type { ForwardedRef, ReactNode } from 'react'
+import React, { useMemo, useRef, useState, forwardRef, useCallback } from 'react'
+import type { EventFilters, FacetRegistry } from '../../../hooks/useEvents'
 import type { SdkEvent } from '../../../sdkEvent'
 import { isRumViewEvent } from '../../../sdkEvent'
+import type { EventListColumn } from './columnUtils'
+import { getColumnTitle, DEFAULT_COLUMNS, includesColumn } from './columnUtils'
 import { EventRow } from './eventRow'
+import { Grid } from './grid'
+import { ColumnDrag } from './columnDrag'
+
+export function EventsList({
+  events,
+  filters,
+  facetRegistry,
+  columns,
+  onColumnsChange,
+}: {
+  events: SdkEvent[]
+  filters: EventFilters
+  facetRegistry: FacetRegistry
+  columns: EventListColumn[]
+  onColumnsChange: (columns: EventListColumn[]) => void
+}) {
+  const headerRowRef = useRef<HTMLDivElement>(null)
+
+  const onAddColumn = useCallback(
+    (column: EventListColumn) => {
+      if (!includesColumn(columns, column)) {
+        onColumnsChange(columns.concat(column))
+      }
+    },
+    [columns]
+  )
 
-export function EventsList({ events, filters }: { events: SdkEvent[]; filters: EventFilters }) {
   return (
-    <Table striped verticalSpacing="xs">
-      <tbody>
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why move from a table to a grid?

Reply:
I reverted back to a table, the grid offers more control over the exact size of columns, but was too much trouble to implement. I want to revisit this one day when I have time :)